civis/tests/test_service_client.py

```diff
@@ -0,0 +1,278 @@
+                    "get": {
+                        "description": "",
+                        "responses": {
+                            "200": {
+                                "description": "Returns a Resource",
+                            }
+                        },
+                        "summary": "Get Resources",
+                        "tags": [
+                            "tag"
+                        ]
+                    },
+                    "patch": {
+                        "description": "",
+                        "parameters": [
+                            {
+                                "description": "The id of the Resource",
+                                "in": "path",
+                                "name": "id",
+                                "required": True,
+                                "type": "integer"
+                            },
+                            {
+                                "description": "The fields and values to edit",
+                                "in": "body",
+                                "name": "body",
+                                "required": True,
+                                "schema": {
+                                    "properties": {
+                                        "field": {
+                                            "description": "a property value",
+                                            "type": "string"
+                                        }
+                                    }
+                                }
+                            }
+                        ],
+                        "responses": {
+                            "200": {
+                                "description": "Edits Resource",
+                            }
+                        },
+                        "summary": "Patch Resources",
+                        "tags": [
+                            "tag"
+                        ]
+                    }
+                }
+            },
+            "swagger": "2.0"
+            }
+
+
+@pytest.fixture
+def mock_operations(mock_swagger):
+    ops_json = mock_swagger["paths"]["/some-resources"]
+    mock_ops_str = str(ops_json).replace('\'', '\"')
+    mock_operations = json.JSONDecoder(object_pairs_hook=OrderedDict).decode(mock_ops_str)  # noqa: E501
+    return mock_operations
+
+
+@mock.patch('civis.service_client.ServiceClient.generate_classes')
+@mock.patch('civis.service_client.civis')
+def test_service_client(mock_civis, classes_mock):
+    mock_client = mock_civis.APIClient()
+    mock_client.services.get.return_value = {"current_url": mock_survey_url}
+    classes_mock.return_value = {}
+
+    sc = ServiceClient(mock_service_id)
+
+    assert sc._session_kwargs == {}
+    assert sc._service_id == mock_service_id
+    assert sc._base_url == mock_survey_url
+    assert sc._root_path == None
+    assert sc._swagger_path == "/endpoints"
```

Comment:
`"/endpoints"` should probably be a constant. then you're tying `_swagger_path` default to a constant value as opposed to an expected string.